.clang-tidy

```diff
@@ -0,0 +1,15 @@
+UseColor: true
+
+# TODO: Discuss if we wanna add some extra checks.
+Checks: >
+    bugprone-*,
+    clang-analyzer-cplusplus*,
+    concurrency-*,
+    cppcoreguidelines-*,
+    google-*,
+    modernize-*,
+    performance-*,
+    readability-*,
+HeaderFilterRegex: '^\.\/eventuals\/.*|^\.\/test\/.*'
```

Comment:
Should we just apply it to all files in this repo?

Reply:
I guess we should, yes.